doc/man3/OSSL_PARAM_dup.pod

```diff
@@ -0,0 +1,58 @@
+=pod
+
+=head1 NAME
+
+OSSL_PARAM_dup, OSSL_PARAM_merge, OSSL_PARAM_free
+- OSSL_PARAM copy functions
+
+=head1 SYNOPSIS
+
+ #include <openssl/params.h>
+
+ OSSL_PARAM *OSSL_PARAM_dup(const OSSL_PARAM *params);
+ OSSL_PARAM *OSSL_PARAM_merge(const OSSL_PARAM *params, const OSSL_PARAM *params1);
+ void OSSL_PARAM_free(OSSL_PARAM *params);
+
+=head1 DESCRIPTION
+
+Algorithm parameters can be exported/imported from/to providers using arrays of
+B<OSSL_PARAM>. The following utility functions allow the parameters to be
+duplicated and merged with other B<OSSL_PARAM> to assist in this process.
+
+OSSL_PARAM_dup() duplicates the parameter array I<params>. This function does a
+deep copy of the data.
+
+OSSL_PARAM_merge() merges the parameter arrays I<params> and I<params1> into a
+new parameter array. If I<params> and I<params1> contain values with the same
+'key' then the value from I<params1> will replace the I<param> value. This
+function does a shallow copy of the parameters. If I<params1> is NULL then a
+copy of I<params> is returned.
```

Comment:
Hmmmm, I suspect that "merge" will set off opinions what it really should do when the same key exists in both arrays.  I can understand why python chose a different name for the corresponding dict method: https://docs.python.org/3/library/stdtypes.html#dict.update

Reply:
We already use _merge_ for properties.  Albeit, with the first replacing the second.

Reply:
If this was only internal, I would care less.  This is a public API, and the question is at least worth raising

Reply:
+1 for `update` vs `merge` (where the former has clear precedence of what is being overwritten in case of same keys, and the latter has divergent interpretations with no clear frontrunner)

Reply:
`update` is very much the wrong term here.

The function is creating a **new** OSSL_PARAM array from two source arrays, it isn't updating an existing one.

Reply:
Ah, good point, @paulidale

Reply:
@levitte, I'm happy for an alternative naming suggestion.  I'm not in the least bit tied to `merge`, I understand the _complications_ with the term.  I'm also unable to suggest a better name.  `combine` is the best I've got which is as bar.

From a purely Boolean logic viewpoint, @slontis has implemented `implies` or `implication`, yet neither are suitable.

Reply:
If you're adding in things that don't exist in the first param set, then maybe `add` works?